app/src/main/java/com/xmartlabs/scasas/criminalintent/ui/DatePickerFragment.java

```diff
@@ -0,0 +1,97 @@
+package com.xmartlabs.scasas.criminalintent.ui;
+
+import android.app.Activity;
+import android.app.Dialog;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.v4.app.DialogFragment;
+import android.support.v7.app.AlertDialog;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.DatePicker;
+
+import com.xmartlabs.scasas.criminalintent.R;
+
+import java.util.Calendar;
+import java.util.Date;
+import java.util.GregorianCalendar;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+
+/**
+ * Created by scasas on 2/16/17.
+ */
+public class DatePickerFragment extends DialogFragment {
+  private static final String ARG_DATE = "date";
+  public static final String EXTRA_DATE = DatePickerFragment.class.getCanonicalName() + ".date";
+
+  @BindView(R.id.dialog_date_picker)
+  DatePicker datePicker;
+
+  @NonNull
+  @Override
+  public Dialog onCreateDialog(Bundle savedInstanceState) {
+    View view = inflateView();
+    ButterKnife.bind(this, view);
+    setupDatePicker();
+    return new AlertDialog.Builder(getActivity())
+        .setView(view)
+        .setTitle(R.string.date_picker_title)
+        .setNegativeButton(android.R.string.cancel, null)
+        .setPositiveButton(android.R.string.ok, onClickAction())
+        .create();
+  }
+
+  private View inflateView() {
+    return LayoutInflater.from(getActivity())
+        .inflate(R.layout.dialog_date, null);
+  }
+
+  public static DatePickerFragment newInstance(Date date) {
+    Bundle arguments = new Bundle();
+    arguments.putSerializable(ARG_DATE, date);
+    DatePickerFragment fragment = new DatePickerFragment();
+    fragment.setArguments(arguments);
+    return fragment;
+  }
+
+  private void setupDatePicker() {
+    int[] date = getIntDateFromCall();
+    datePicker.init(date[0], date[1], date[2], null);
+  }
+
+  private int[] getIntDateFromCall() {
+    Date date = (Date) getArguments().getSerializable(ARG_DATE);
+    Calendar calendar = Calendar.getInstance();
+    calendar.setTime(date);
+    int year = calendar.get(Calendar.YEAR);
+    int month = calendar.get(Calendar.MONTH);
+    int day = calendar.get(Calendar.DAY_OF_MONTH);
+    return new int[] {year, month, day};
```

Comment:
you shouldn't use a array, a list is better. However, you shouldn't returns the dates in a list/array, it's not mantainable.